### PR TITLE
CBL-3948: [test] Update test to validate more date formats

### PR DESCRIPTION
**Pull request by jayahariv**
* Already done different formats and parsing logic
* Once parsed, will save to LiteCore and retrive when sendWebSocketRequest

### CODE REVIEW

File: Objective-C/Tests/ReplicatorTest+Main.m [hunk 1]
```diff
@@ -1538,10 +1538,22 @@ - (void) testWebSocketParseCookie {
...
+        @[@"a1=b1;expires=b1,2,a3=b3;Expires=b3,4",
+          @[@"a1=b1;expires=b1,2", @"a3=b3;Expires=b3,4"]],
+        
+        // RFC 822, updated by RFC 1123
+        @[@"a1=b1;expires=Sun, 06 Nov 1994 08:49:37 GMT;Path=/",
+          @[@"a1=b1;expires=Sun, 06 Nov 1994 08:49:37 GMT;Path=/"]],
+        
+        // RFC 850, obsoleted by RFC 1036
+        @[@"a1=b1;expires=Sunday, 06-Nov-94 08:49:37 GMT;Path=/",
+          @[@"a1=b1;expires=Sunday, 06-Nov-94 08:49:37 GMT;Path=/"]],
```
Comment (borrrden) [inline Objective-C/Tests/ReplicatorTest+Main.m:1551]: Weird, we actually left this format out.  I don't see the GCLB format in here, is it already included?

Comment (jayahariv) [reply Objective-C/Tests/ReplicatorTest+Main.m:1551]: Above format is similar right? 

```
GCLB=<token>; path=/; HttpOnly; expires=Tue, 22-Nov-2022 07:21:38 GMT
==> 
a1=b1;expires=Sunday, 06-Nov-94 08:49:37 GMT;Path=/"
```

Only difference we are validating is the three different date formats Pasin added as a [comment](https://issues.couchbase.com/browse/CBL-3941?focusedCommentId=644330&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-644330) right?

Comment (jayahariv) [reply Objective-C/Tests/ReplicatorTest+Main.m:1551]: This are the three date formats Pasin mentioned in the [comment ](https://issues.couchbase.com/browse/CBL-3941?focusedCommentId=644330&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-644330)

Comment (borrrden) [reply Objective-C/Tests/ReplicatorTest+Main.m:1551]: We didn't actually do RFC 850

**Review by borrrden — APPROVED** [review]

**Pull request merged**